osutil/mountentry_linux.go

```diff
@@ -21,106 +21,11 @@ package osutil
 
 import (
 	"fmt"
-	"math"
-	"os"
-	"regexp"
 	"strconv"
 	"strings"
 	"syscall"
 )
 
-// MountEntry describes an /etc/fstab-like mount entry.
-//
-// Fields are named after names in struct returned by getmntent(3).
-//
-// struct mntent {
-//     char *mnt_fsname;   /* name of mounted filesystem */
-//     char *mnt_dir;      /* filesystem path prefix */
-//     char *mnt_type;     /* mount type (see Mntent.h) */
-//     char *mnt_opts;     /* mount options (see Mntent.h) */
-//     int   mnt_freq;     /* dump frequency in days */
-//     int   mnt_passno;   /* pass number on parallel fsck */
-// };
-type MountEntry struct {
-	Name    string
-	Dir     string
-	Type    string
-	Options []string
-
-	DumpFrequency   int
-	CheckPassNumber int
-}
-
-func equalStrings(a, b []string) bool {
-	if len(a) != len(b) {
-		return false
-	}
-	for i := 0; i < len(a); i++ {
-		if a[i] != b[i] {
-			return false
-		}
-	}
-	return true
-}
-
-// Equal checks if one entry is equal to another
-func (e *MountEntry) Equal(o *MountEntry) bool {
-	return (e.Name == o.Name && e.Dir == o.Dir && e.Type == o.Type &&
-		equalStrings(e.Options, o.Options) && e.DumpFrequency == o.DumpFrequency &&
-		e.CheckPassNumber == o.CheckPassNumber)
-}
-
-// escape replaces whitespace characters so that getmntent can parse it correctly.
-var escape = strings.NewReplacer(
-	" ", `\040`,
-	"\t", `\011`,
-	"\n", `\012`,
-	"\\", `\134`,
-).Replace
-
-// unescape replaces escape sequences used by setmnt with whitespace characters.
-var unescape = strings.NewReplacer(
-	`\040`, " ",
-	`\011`, "\t",
-	`\012`, "\n",
-	`\134`, "\\",
-).Replace
-
-// Escape returns the given path with space, tab, newline and forward slash escaped.
-func Escape(path string) string {
-	return escape(path)
-}
-
-// Unescape returns the given path with space, tab, newline and forward slash unescaped.
-func Unescape(path string) string {
-	return unescape(path)
-}
-
-func (e MountEntry) String() string {
-	// Name represents name of the device in a mount entry.
-	name := "none"
-	if e.Name != "" {
-		name = escape(e.Name)
-	}
-	// Dir represents mount directory in a mount entry.
-	dir := "none"
-	if e.Dir != "" {
-		dir = escape(e.Dir)
-	}
-	// Type represents file system type in a mount entry.
-	fsType := "none"
-	if e.Type != "" {
-		fsType = escape(e.Type)
-	}
-	// Options represents mount options in a mount entry.
-	options := "defaults"
-	if len(e.Options) != 0 {
-		options = escape(strings.Join(e.Options, ","))
-	}
-	return fmt.Sprintf("%s %s %s %s %d %d",
-		name, dir, fsType, options, e.DumpFrequency, e.CheckPassNumber)
-}
-
 // ParseMountEntry parses a fstab-like entry.
 func ParseMountEntry(s string) (MountEntry, error) {
```

Comment:
this could go in the shared code,  it doesn't seem to use syscall, am I missing something? is the worry that is linux specific but then likely MountEntry itself is as well?